nanomq/apps/broker.c

```diff
@@ -431,41 +430,34 @@ server_cb(void *arg)
 			msg_infos = work->pipe_ct->msg_infos;
 
 			debug_msg("total pipes: %ld", cvector_size(msg_infos));
-			//TODO encode abstract msg only
 			if (cvector_size(msg_infos))
-				encode_pub_message(
-				    smsg, work, PUBLISH, msg_info);
-			for (int i = 0; i < cvector_size(msg_infos); ++i) {
-				msg_info = &msg_infos[i];
-				nng_msg_clone(smsg);
-				work->pid.id = msg_info->pipe;
-				nng_msg_set_pipe(smsg, work->pid);
-				work->msg = NANO_NNI_LMQ_PACKED_MSG_QOS(smsg, msg_info->qos);
-				nng_aio_set_prov_data(work->aio, (void *)msg_info->retain);
-				nng_aio_set_msg(work->aio, work->msg);
-				nng_ctx_send(work->ctx, work->aio);
-				nng_aio_set_prov_data(work->aio, NULL);
-			}
+				if (encode_pub_message(smsg, work, PUBLISH, msg_info))
```

Comment:
Actually, encode_pub_message always returns true.

Reply:
not in this PR